OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -141,15 +147,20 @@ public Armament(Actor self, ArmamentInfo info)
 
 		public virtual WDist MaxRange()
 		{
-			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, rangeModifiers));
+			return new WDist(Util.ApplyPercentageModifiers(Weapon.Range.Length, initialRangeModifiers));
 		}
 
 		protected override void Created(Actor self)
 		{
 			turret = self.TraitsImplementing<Turreted>().FirstOrDefault(t => t.Name == Info.Turret);
 			ammoPool = self.TraitsImplementing<AmmoPool>().FirstOrDefault(la => la.Info.Name == Info.AmmoPoolName);
 			coords = self.Trait<BodyOrientation>();
-			rangeModifiers = self.TraitsImplementing<IRangeModifier>().ToArray().Select(m => m.GetRangeModifier());
+			notifyBurstComplete = self.TraitsImplementing<INotifyBurstComplete>().ToArray();
```

Comment:
... ~the game has fully loaded~ all of their traits have been constructed and added to them (can be looked up).